### DISCUSSION TITLE
Allow dragging of component only through the toolbar

### DISCUSSION THREAD

**Discussion by Agobin** [GrapesJS/grapesjs#3708] [Q&A]
I recently upgraded my version of grapes and I realized dragging of components works differently in the newer version. In the previous version, you could only drag a component by using the toolbar icon. In the latest version, when you press down on a component and move the mouse, it starts moving the component.

How can I prevent users from moving a component by pressing down on the component? To make moving components possible only by using the toolbar.

Thank you.

**Comment by Agobin** [comment]
I realized executing the code below seems to work. Just wondering if this is the best way to accomplish this.

editor.on('run:tlb-move:before', opts => opts.abort = 1; )

**Reply by artf** [reply]
There is an option to disable it on all components
```js
grapesjs.init({
  ...
  domComponents: { draggableComponents: false },
})
```

**Reply by Agobin** [reply]
Thank you @artf, this is the perfect solution.
